homeassistant/components/music_assistant/config_flow.py

```diff
@@ -0,0 +1,168 @@
+    CannotConnect,
+    InvalidServerVersion,
+    MusicAssistantClientException,
+)
+from music_assistant.common.models.api import ServerInfoMessage
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components import zeroconf
+from homeassistant.config_entries import ConfigFlowResult
+from homeassistant.const import CONF_URL
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import AbortFlow
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers import aiohttp_client
+
+from .const import DOMAIN, LOGGER
+
+DEFAULT_URL = "http://mass.local:8095"
+DEFAULT_TITLE = "Music Assistant"
+
+
+def get_manual_schema(user_input: dict[str, Any]) -> vol.Schema:
+    """Return a schema for the manual step."""
+    default_url = user_input.get(CONF_URL, DEFAULT_URL)
+    return vol.Schema(
+        {
+            vol.Required(CONF_URL, default=default_url): str,
+        }
+    )
+
+
+async def get_server_info(hass: HomeAssistant, url: str) -> ServerInfoMessage:
+    """Validate the user input allows us to connect."""
+    async with MusicAssistantClient(
+        url, aiohttp_client.async_get_clientsession(hass)
+    ) as client:
+        if TYPE_CHECKING:
+            assert client.server_info is not None
+        return client.server_info
+
+
+# ruff: noqa: ARG002
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for MusicAssistant."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Set up flow instance."""
+        self.server_info: ServerInfoMessage | None = None
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        return await self.async_step_manual()
+
+    async def async_step_manual(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle a manual configuration."""
+        if user_input is None:
+            return self.async_show_form(  # type: ignore[no-any-return]
+                step_id="manual", data_schema=get_manual_schema({})
+            )
+
+        errors = {}
+        try:
+            self.server_info = await get_server_info(self.hass, user_input[CONF_URL])
+            await self.async_set_unique_id(self.server_info.server_id)
+        except CannotConnect:
+            errors["base"] = "cannot_connect"
+        except InvalidServerVersion:
+            errors["base"] = "invalid_server_version"
+        except MusicAssistantClientException:  # pylint: disable=broad-except
+            LOGGER.exception("Unexpected exception")
+            errors["base"] = "unknown"
+        else:
+            return await self._async_create_entry_or_abort()
+
+        return self.async_show_form(  # type: ignore[no-any-return]
+            step_id="manual", data_schema=get_manual_schema(user_input), errors=errors
+        )
```

Comment:
I personally like to inverse the if statement, to make it `if user_input is not None` and then just have one call to `async_show_form`

Reply:
What if statement ?

Reply:
The if statement at the top

Reply:
ah I see. I guess that is personal preference. 
I'm a fan of guards and less indentation but as the resulting action is indeed the same, I agree we should adjust this